### DISCUSSION TITLE
Cannot decrypt file with corrected private key

### DISCUSSION THREAD

**Discussion by Xing1P** [openpgpjs/openpgpjs#1536] [Q&A]
Node v: 16.13.0
Openpgp v: 5.2.1
I got decrypt error " Error decrypting message: Decryption error " but can decrypt this file with [PGPTool](https://pgptool.github.io)
Please help
- Decrypt got error :  but this file I can decrypt in PGPTool
![message de error](https://user-images.githubusercontent.com/13740546/173782984-095259b1-38e8-4d33-abf7-2cf0b97c21cd.png)
- Decrypt not error:
![message de not error](https://user-images.githubusercontent.com/13740546/173783539-e05af927-15a6-4439-8019-b0273342119b.png)
I have note when message no session key it cannot decrypt, It's right?
[key and encrypted file.zip](https://github.com/openpgpjs/openpgpjs/files/8907394/key.and.encrypted.file.zip)

This my sample 
`const fileArray = await blob.arrayBuffer();
                const uint8 = new Uint8Array(fileArray);
                const privateKeyObj = await openpgp.readPrivateKey({ armoredKey: privateKey });
                const message = await openpgp.readMessage({binaryMessage: uint8});
                console.log("decryptFile-key", privateKeyObj);
                console.log("decryptFile-message", message);
                const { data: decrypted} = await openpgp.decrypt({
                    message: message,
                    decryptionKeys: privateKeyObj,
                    format: 'binary',
                    config: { allowInsecureDecryptionWithSigningKeys: true }
                });`

**Comment by larabr** [comment]
Hi @Xing1P , decryption works for me (without any `config` option), but I manually converted the binary file to armored data using GPG. So I think the problem is with the way you are reading the file (make sure the encoding is correct, for instance).

**Reply by Xing1P** [reply]
Thanks for your reply. 
What's your manually converted the binary file to armored data? you can tell the way you are, so I can test it to make clear.
I already try without config too but still not working.
My sample like this still decrypt error:
`const message = await openpgp.readMessage({binaryMessage:
                    fs.createReadStream('C:/.../encryptedImage.gpg')}
                  )
                const { data: decrypted} = await openpgp.decrypt({
                    message: message,
                    decryptionKeys: privateKeyObj,
                    format: 'binary'
                });`

**Reply by larabr** [reply]
You can use `gpg --enarmor < file` and then replace the delimiters from `-----BEGIN/END PGP ARMORED FILE-----` to `-----BEGIN/END PGP MESSAGE-----`

**Reply by Xing1P** [reply]
I think it problem in packetlist.js cannot get sessionkey.

**Reply by Xing1P** [reply]
Now it work when after do this, please help explain.
Note: I'm using in Electron v13
# Before: not work for some file for me
- npm install openpgp@5.3.0 
- const openpgp = require('openpgp')
# Work now
- clone openpgp repo v5.3.0
- and then get openpgp.js from dist and import to my project
- <script src="openpgp.js"></script>
